Sources/Load.php

```diff
@@ -3846,17 +3846,32 @@ function set_avatar_data($data = array())
  */
 function get_auth_secret()
 {
-	global $auth_secret, $sourcedir, $smcFunc;
+	global $context, $auth_secret, $sourcedir, $smcFunc, $db_last_error, $txt;
 
 	if (empty($auth_secret))
 	{
 		$auth_secret = bin2hex($smcFunc['random_bytes'](32));
 
 		// It is important to store this in Settings.php, not the database.
 		require_once($sourcedir . '/Subs-Admin.php');
-		updateSettingsFile(array('auth_secret' => $auth_secret));
+
+		// Did this fail?  If so, we should alert, log and set a static value.
+		if (!updateSettingsFile(array('auth_secret' => $auth_secret)))
+		{
+			$context['auth_secret_missing'] = true;
+			$auth_secret = 'none';
```

Comment:
```suggestion
			$auth_secret = hash_file('sha256', $boarddir . '/Settings.php');
```

This will keep the $auth_secret value unique to each forum, even if not random. Since the issue is that Settings.php is not writable, this approach should keep the value stable at least until the admin fixes the permissions problem.